### PR TITLE
Fix error when reusing the encoder for decode/encode 

### PR DESCRIPTION
**Pull request by DomAmato**
There was a bug where it did not reset the length value after decoding a packet which lead to it raising an encoding error due to the length not matching.

### CODE REVIEW

File: smpp/pdu/pdu_types.py [hunk 1]
```diff
@@ -126,7 +126,18 @@ def __repr__(self):
 LanguageIndicator = Enum('LanguageIndicator', list(constants.language_indicator_name_map.keys()))
 DisplayTime = Enum('DisplayTime', list(constants.display_time_name_map.keys()))
 MsAvailabilityStatus = Enum('MsAvailabilityStatus', list(constants.ms_availability_status_name_map.keys()))
-NetworkErrorCode = Enum('NetworkErrorCode', list(constants.network_error_code_name_map.keys()))
+
+
+NetworkErrorCodeNetworkType = Enum('NetworkErrorCodeNetworkType', list(constants.network_error_code_name_map.keys()))
+
+NetworkErrorCodeBase = namedtuple('NetworkErrorCode', 'networkType, value')
```
Comment (HologramInterview) [inline smpp/pdu/pdu_types.py]:
```suggestion
NetworkErrorCodeBase = namedtuple('NetworkErrorCode', 'networkType', 'value')
```

Comment (andrewHologram) [inline smpp/pdu/pdu_types.py:133]:
```suggestion
NetworkErrorCodeBase = namedtuple('NetworkErrorCode', 'networkType', 'value')
```

Comment (DomAmato) [reply smpp/pdu/pdu_types.py]: that is the wrong syntax for named tuples: https://docs.python.org/3/library/collections.html#collections.namedtuple

Comment (andrewHologram) [reply smpp/pdu/pdu_types.py]: Oh interesting I thought it was a typo.

**Pull request merged**